packages/grafana-ui/src/components/Segment/SegmentSelect.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useRef, useEffect, useState } from 'react';
+import { css, cx } from 'emotion';
+import useClickAway from 'react-use/lib/useClickAway';
+import { SelectableValue } from '@grafana/data';
+import { Select } from '../Select/Select';
+import { OptionType } from './';
+
+export interface Props<T> {
+  options: OptionType<T>;
+  onChange: (value: T) => void;
+  onClickOutside: () => void;
+  width: number;
+  noOptionsMessage?: string;
+}
+
+export function SegmentSelect<T>({
+  options = [],
+  onChange,
+  onClickOutside,
+  width,
+  noOptionsMessage = '',
+}: React.PropsWithChildren<Props<T>>) {
+  const ref = useRef(null);
+  const [optionTypes, setOptionsTypes] = useState<Array<SelectableValue<T>>>([]);
+
+  useEffect(() => {
+    const selectOptions = Array.isArray(options)
+      ? options
+      : Object.entries(options).map(([key, values]: [string, Array<OptionType<T>>]) => {
```

Comment:
The use of OptionType here and this part makes here makes me confused, is this not already handled in Select component? why do we need to do anything here? Should the options not already be in grouped already? Why does this have different type / way of dealing with grouped options than main Select.tsx? 

Or do you think this makes working with grouped options easier and should be part of main Select.tsx?

Reply:
haha I have no clue...I think I thought that this is how Select used to work a year ago, so now I didn't check the API properly. I'll remove OptionType.